tests/dataimportcron_test.go

```diff
@@ -0,0 +1,225 @@
+
+		if schedule == scheduleEveryMinute {
+			var cronjob *v1beta1.CronJob
+			By("Verify cronjob was created")
+			Eventually(func() bool {
+				cronjob, err = f.K8sClient.BatchV1beta1().CronJobs(f.CdiInstallNs).Get(context.TODO(), controller.GetCronJobName(cron), metav1.GetOptions{})
+				return err == nil && len(cronjob.Status.Active) > 0
+			}, dataImportCronTimeout, pollingInterval).Should(BeTrue())
+			Expect(err).ToNot(HaveOccurred())
+
+			By(fmt.Sprintf("Find pod of job %s", cronjob.Status.Active[0].Name))
+			Eventually(func() bool {
+				_, err = utils.FindPodByPrefix(f.K8sClient, f.CdiInstallNs, cronjob.Status.Active[0].Name, "")
+				return err == nil
+			}, dataImportCronTimeout, pollingInterval).Should(BeTrue())
+		}
+
+		var lastImportDv, currentImportDv string
+		for i := 0; i < repeat; i++ {
+			if repeat > 1 {
+				// Emulate source update using digests from https://quay.io/repository/kubevirt/cirros-container-disk-demo?tab=tags
+				digest := []string{
+					"sha256:68b44fc891f3fae6703d4b74bcc9b5f24df8d23f12e642805d1420cbe7a4be70",
+					"sha256:90e064fca2f47eabce210d218a45ba48cc7105b027d3f39761f242506cad15d6",
+				}[i%2]
+
+				By(fmt.Sprintf("Update source desired digest to %s", digest))
+				Eventually(func() bool {
+					cron, err = f.CdiClient.CdiV1beta1().DataImportCrons(f.Namespace.Name).Get(context.TODO(), cron.Name, metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					if cron.Annotations == nil {
+						cron.Annotations = make(map[string]string)
+					}
+					cron.Annotations[controller.AnnSourceDesiredDigest] = digest
+					cron, err = f.CdiClient.CdiV1beta1().DataImportCrons(f.Namespace.Name).Update(context.TODO(), cron, metav1.UpdateOptions{})
+					return err == nil
+				}, dataImportCronTimeout, pollingInterval).Should(BeTrue())
+			}
+			By("Wait for CurrentImports DataVolumeName update")
+			Eventually(func() bool {
+				cron, err = f.CdiClient.CdiV1beta1().DataImportCrons(f.Namespace.Name).Get(context.TODO(), cron.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				if cron.Status.CurrentImports == nil {
+					return false
+				}
+				currentImportDv = cron.Status.CurrentImports[0].DataVolumeName
+				return currentImportDv != "" && currentImportDv != lastImportDv
+			}, dataImportCronTimeout, pollingInterval).Should(BeTrue())
+
+			lastImportDv = currentImportDv
+
+			By(fmt.Sprintf("Verify pvc was created %s", currentImportDv))
+			_, err = utils.WaitForPVC(f.K8sClient, cron.Namespace, currentImportDv)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Wait for import completion")
+			err = utils.WaitForDataVolumePhase(f.CdiClient, cron.Namespace, cdiv1.Succeeded, currentImportDv)
+			Expect(err).ToNot(HaveOccurred(), "Datavolume not in phase succeeded in time")
+
+			By("Verify datasource was updated")
+			Eventually(func() bool {
+				datasource, err := f.CdiClient.CdiV1beta1().DataSources(f.Namespace.Name).Get(context.TODO(), cron.Spec.ManagedDataSource, metav1.GetOptions{})
```

Comment:
You have a flaky test noting that the datasource doesn't exist yet sometimes. This is because after the DV is complete the controller then creates the datasource. But we might get here before that actually happens. So I would check if `errors.IsNotFound(err)` is true, and if so return false, so we retry in a few seconds.

Reply:
fixed